doc/release-notes-31384.md

```diff
@@ -0,0 +1,16 @@
+- Node and Mining
+
+---
+
+- PR #31384 fixed an issue where the coinbase transaction weight was being reserved in two different places.
+  The fix ensures that the reservation happens in a single place.
+
+- This PR also introduces a new startup option, `-maxcoinbaseweight` (default: `8,000` weight units).
+  This specifies how many weight units to reserve for the coinbase transaction.
+- Upgrade note: the default of `-maxcoinbaseweight` ensures backward compatibility for users who did
+  not override `-blockmaxweight`.
```

Comment:
2fb833e69627a8f9b1c4b14b60d1df9903adf210

You might want to mention the old default of `3996000` here. I was staring at `blockmaxweight=4000000` on the following line for a minute trying to figure out what actually changed.

Reply:
Good point, maybe add:

```
The previous `-blockmaxweight` default was `3,996,000`. The block template construction code added an additional `4,000` padding. The new default `-maxcoinbaseweight` value was chosen to maintain that behaviour.